docs/csharp/tutorials/console-webapiclient.md

```diff
@@ -1,388 +1,354 @@
-<xref:System.Net.Http.HttpClient.GetStringAsync(System.String)> in your `ProcessRepositories` method with the following lines:
+1. Add two `using` directives at the top of the file:
 
-```csharp
-var streamTask = client.GetStreamAsync("https://api.github.com/orgs/dotnet/repos");
-var repositories = await JsonSerializer.DeserializeAsync<List<Repository>>(await streamTask);
-```
+   ```csharp
+   using System.Net.Http;
+   using System.Net.Http.Headers;
+   ```
 
-You're using new namespaces, so you'll need to add it at the top of the file as well:
+1. Build the application and run it.
 
-```csharp
-using System.Collections.Generic;
-using System.Text.Json;
-```
+   ```dotnetcli
+   dotnet run
+   ```
 
-Notice that you're now using <xref:System.Net.Http.HttpClient.GetStreamAsync(System.String)> instead of <xref:System.Net.Http.HttpClient.GetStringAsync(System.String)>. The serializer
-uses a stream instead of a string as its source. Let's explain a couple features of the C#
-language that are being used in the second line of the preceding code snippet. The first argument to <xref:System.Text.Json.JsonSerializer.DeserializeAsync%60%601(System.IO.Stream,System.Text.Json.JsonSerializerOptions,System.Threading.CancellationToken)?displayProperty=nameWithType> is an
-`await` expression. (The other two parameters are optional and are omitted in the code snippet.) Await expressions can appear almost anywhere in your code, even though
-up to now, you've only seen them as part of an assignment statement. The `Deserialize` method is *generic*, which means you must supply type arguments for what kind of objects should be created from the JSON text. In this example, you're deserializing to a `List<Repository>`, which is another generic object, the <xref:System.Collections.Generic.List%601?displayProperty=nameWithType>. The `List<>` class stores a collection of objects. The type argument declares the type of objects stored in the `List<>`. The JSON text represents a collection of repo objects, so the type argument is `Repository`.
+   There is no build warning, because the `ProcessRepositories` now does contain an `await` operator.
 
-You're almost done with this section. Now that you've converted the JSON to C# objects, let's display
-the name of each repository. Replace the lines that read:
+   The output is a long display of JSON text.
 
-```csharp
-var msg = await stringTask;   //**Deleted this
-Console.Write(msg);
-```
+## Deserialize the JSON Result
 
-with the following:
+The following steps convert the JSON response into C# objects. You use the <xref:System.Text.Json.JsonSerializer?displayProperty=nameWithType> class to deserialize JSON into objects.
 
-```csharp
-foreach (var repo in repositories)
-    Console.WriteLine(repo.name);
-```
+1. Create a file named *repo.cs* and put the following code in it. This code defines a class to represent the JSON object returned from the GitHub API:
```

Comment:
Your version is fine. I write it like the following:
```suggestion
1. Create a file named *repo.cs* and add the following code:
```

Move this after
The preceding code defines a class to represent the JSON object returned from the GitHub API: